streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java

```diff
@@ -203,6 +203,171 @@ private void setupGetSensorTest(final Metrics metrics,
         replay(metrics);
     }
 
+
+    private void setupGetNewSensorTest(final Metrics metrics,
+                                       final String level,
+                                       final RecordingLevel recordingLevel) {
+        final String fullSensorName =
+            INTERNAL_PREFIX + SENSOR_PREFIX_DELIMITER + level + SENSOR_NAME_DELIMITER + sensorName1;
+        expect(metrics.getSensor(fullSensorName)).andStubReturn(null);
+        final Sensor[] parents = {};
+        expect(metrics.sensor(fullSensorName, recordingLevel, parents)).andReturn(sensor);
+        replay(metrics);
+    }
+
+    private void setupGetExistingSensorTest(final Metrics metrics,
+                                            final String level) {
+        final String fullSensorName =
+            INTERNAL_PREFIX + SENSOR_PREFIX_DELIMITER + level + SENSOR_NAME_DELIMITER + sensorName1;
+        expect(metrics.getSensor(fullSensorName)).andStubReturn(sensor);
+        replay(metrics);
+    }
+
+    @Test
+    public void shouldGetNewThreadLevelSensor() {
```

Comment:
This test replaces `shouldGetThreadLevelSensor()`. Thus, you can safely remove `shouldGetThreadLevelSensor()`.